cache/transaction_registry.go

```diff
@@ -21,6 +22,9 @@ type TransactionRegistry interface {
 	Status(key *Key) (TransactionStatus, error)
 }
 
+// transactionEndedTTL amount of time transaction record is kept after being updated
+const transactionEndedTTL = 500 * time.Millisecond
```

Comment:
IMHO it should be configurable because some people will tell you they want 0 msec

Reply:
I was thinking about it to, but it makes few sense to expose it in the configuration because it's a technical detail of the implementation. I'd say instead of exposing it, we could expose a toggle feature flag to disable thundering herd explicitly

Reply:
here is my proposal: let's ship this feature first and let's see how people respond.